rossum.py

```diff
@@ -655,27 +655,27 @@ def resolve_includes(pkgs):
     logger.debug("Resolving includes for: {}".format(', '.join(pkg_names)))
 
     for pkg in pkgs:
-        # TODO: watch out for circular dependencies
+        visited = set()
```

Comment:
A visited-set per workspace may not be the right thing to use in that case. If another package in the source depends on the same packages, those packages will get overlooked in the package that is looked at last. for example just putting the visited set outside of the package loop yields:
```
### Hash ###################

Hash_dir = c:\Users\matt\Dropbox\Programming\Git\Ka-Boost\vendor\Hash\src
Hash_deps = KUnit Strings
Hash_include_flags = /I"c:\Users\matt\Dropbox\Programming\Git\Ka-Boost\vendor\Hash\src\include" /I"c:\Users\matt\Dropbox\Programming\Git\Ka-Boost\vendor\KUnit\src\include" /I"c:\Users\matt\Dropbox\Programming\Git\Ka-Boost\vendor\Strings\src\include" /I"C:\Users\matt\Dropbox\Programming\Git\Ka-Boost\lib\errors\src\include"

build $build_dir\hash.pc: ktrans_pc $Hash_dir\src\hash.kl
  lib_includes = $Hash_include_flags
  description = Hash :: src\hash.kl

build $build_dir\test_hash.pc: ktrans_pc $Hash_dir\src\test\test_hash.kl
  lib_includes = $Hash_include_flags
  description = Hash :: src\test\test_hash.kl



### KUnit ###################

KUnit_dir = c:\Users\matt\Dropbox\Programming\Git\Ka-Boost\vendor\KUnit\src
KUnit_deps = Strings
KUnit_include_flags = 

build $build_dir\kunit.pc: ktrans_pc $KUnit_dir\src\kunit.kl
  lib_includes = $KUnit_include_flags
  description = KUnit :: src\kunit.kl

build $build_dir\test_kunit.pc: ktrans_pc $KUnit_dir\src\test\test_kunit.kl
  lib_includes = $KUnit_include_flags
  description = KUnit :: src\test\test_kunit.kl
```

Where kunit fails in ktransw.

Also, the way the package manifests are set now the depends package do not have any version association with them. The only thing I managed to do with the dependency tree code is to use the most current version if there were multiple versions found.

I think it is a good idea to open it up as an issue and try to tackle it in a future commit/PR. It will take a some tweaking to get the manifest into a format of :

```
"depends" : [
    "Strings<=1.2.0"
  ]
```
if you think that is appropriate.